File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;
using DevExpress.XtraReports.UI;
namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {

            RichEditDocumentServer server = new RichEditDocumentServer();
            server.LoadDocument("fish.rtf");
            DocumentRange[] ranges = server.Document.FindAll(DevExpress.Office.Characters.PageBreak.ToString(), DevExpress.XtraRichEdit.API.Native.SearchOptions.None);
            DocumentPosition dp = server.Document.Paragraphs[0].Range.Start;
            
            List<MyRtfObject> collection = new List<MyRtfObject>();
            foreach(DocumentRange dr in ranges) {
                DocumentRange tmpRange = server.Document.CreateRange(dp, dr.Start.ToInt() - dp.ToInt());
                collection.Add(new MyRtfObject() { RtfSplitContent = server.Document.GetRtfText(tmpRange) });
                dp = dr.End;
            }
            DocumentRange tmpRange2 = server.Document.CreateRange(dp, server.Document.Paragraphs[server.Document.Paragraphs.Count-1].Range.End.ToInt() - dp.ToInt());
            collection.Add(new MyRtfObject() { RtfSplitContent = server.Document.GetRtfText(tmpRange2) });

            XtraReport1 report = new XtraReport1();
            report.DataSource = collection;
            report.ShowPreviewDialog();
        }
    }

    public class MyRtfObject {
        public string RtfSplitContent { get; set; }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native
Imports DevExpress.XtraReports.UI
Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click

			Dim server As New RichEditDocumentServer()
			server.LoadDocument("fish.rtf")
			Dim ranges() As DocumentRange = server.Document.FindAll(DevExpress.Office.Characters.PageBreak.ToString(), DevExpress.XtraRichEdit.API.Native.SearchOptions.None)
			Dim dp As DocumentPosition = server.Document.Paragraphs(0).Range.Start

			Dim collection As New List(Of MyRtfObject)()
			For Each dr As DocumentRange In ranges
				Dim tmpRange As DocumentRange = server.Document.CreateRange(dp, dr.Start.ToInt() - dp.ToInt())
				collection.Add(New MyRtfObject() With {.RtfSplitContent = server.Document.GetRtfText(tmpRange)})
				dp = dr.End
			Next dr
			Dim tmpRange2 As DocumentRange = server.Document.CreateRange(dp, server.Document.Paragraphs(server.Document.Paragraphs.Count-1).Range.End.ToInt() - dp.ToInt())
			collection.Add(New MyRtfObject() With {.RtfSplitContent = server.Document.GetRtfText(tmpRange2)})

			Dim report As New XtraReport1()
			report.DataSource = collection
			report.ShowPreviewDialog()
		End Sub
	End Class

	Public Class MyRtfObject
		Private privateRtfSplitContent As String
		Public Property RtfSplitContent() As String
			Get
				Return privateRtfSplitContent
			End Get
			Set(ByVal value As String)
				privateRtfSplitContent = value
			End Set
		End Property
	End Class
End Namespace
